cfgmgr/tunnelmgr.cpp

```diff
@@ -0,0 +1,88 @@
+#include <algorithm>
+#include <regex>
+#include <sstream>
+#include <string>
+#include <net/if.h>
+
+#include "logger.h"
+#include "tunnelmgr.h"
+
+using namespace std;
+using namespace swss;
+
+TunnelMgr::TunnelMgr(DBConnector *cfgDb, DBConnector *appDb, const vector<std::string> &tables) :
+        Orch(cfgDb, tables),
+        m_appIpInIpTunnelTable(appDb, APP_TUNNEL_DECAP_TABLE_NAME)
+{
+}
+
+void TunnelMgr::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        bool task_result = false;
+
+        KeyOpFieldsValuesTuple t = it->second;
+        const vector<FieldValueTuple>& data = kfvFieldsValues(t);
+
+        const std::string & op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            for (auto idx : data)
+            {
+                const auto &field = fvField(idx);
+                const auto &value = fvValue(idx);
+                if (field == "tunnel_type")
+                {
+                    if (value == "IPINIP")
+                    {
+                        task_result = doIpInIpTunnelTask(t);
+                    }
+                }
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            /* TODO: Handle Tunnel delete for other tunnel types */
```

Comment:
Can we add a check for value == "IPINIP" before calling doIpInIpTunnelTask(t)?  This way it is in parity with how the "SET_COMMAND" is handled that do check for the tunnel type.

Reply:
In the del operation we will not get the value